llvm/test/CodeGen/AMDGPU/srem.ll

```diff
@@ -2701,12 +2697,10 @@ define amdgpu_kernel void @srem_v2i64(ptr addrspace(1) %out, ptr addrspace(1) %i
 ; GCN-NEXT:    s_waitcnt vmcnt(0)
 ; GCN-NEXT:    v_readfirstlane_b32 s11, v5
 ; GCN-NEXT:    v_readfirstlane_b32 s10, v4
-; GCN-NEXT:    s_or_b64 s[6:7], s[10:11], s[8:9]
```

Comment:
The transformation is invalid in this case.  Must ensure that the result of the non-zero def (i.e. s_or_b64) is only s7 in this case.  Since it isn't, pierceThroughRegSequence should return null in this case.

Reply:
Hi @LU-JOHN yes this is broken.  Thanks.  I'll take a look.